### DISCUSSION TITLE
customizing the window's header

### DISCUSSION THREAD

**Discussion by npolyak** [AvaloniaUI/Avalonia#6170] [Q&A] [answered]
Is there a way to customize the window's header, e.g. changing its Background, removing or changing the maximize/minimize/kill buttons, placing some extra controls within the header regions?

The following 3 properties allow removing the usual chrome from the window on Windows, while keeping the ability to move and resize the window:

        ExtendClientAreaToDecorationsHint="True"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaTitleBarHeightHint="25"

but it does not work on every platform - e.g. ubuntu 20.4:
![image](https://user-images.githubusercontent.com/2833722/124220908-a4eb4e00-dacc-11eb-8daf-a99a196fe8b3.png)

If I set SystemDecorations="BorderOnly" I remove the header, but I also lose the ability to resize and move the window on Ubuntu.

Please, give me advice of how to customize the header on Linux and MAC (if at all possible)

**Comment by maxkatz6** [comment]
@kekekeks @danwalmsley

**Comment by kekekeks** [comment, marked as answer]
Since X11 draws window frame in a separate process, there is nowhere for us to extend. See manual decorations example here: https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/DecoratedWindow.xaml#L1

Also note that manual/client decorations _break user experience_ for most non-GNOME based desktop environments (e. g. KDE and tiled window managers), so it's recommended to check for XDG_CURRENT_DESKTOP environment variable before using those.

**Reply by npolyak** [reply]
Yes, DecoratedWindow is something I need. thanks.
